WebApp-your-API/B2C/README-Incremental.md

```diff
@@ -0,0 +1,292 @@
+
+1. Open the `Service\appsettings.json` file.
+1. Find the key `Instance` and replace the value with your tenant name. For example, `https://fabrikam.b2clogin.com`
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of the `ToDoListService-aspnetcore` application copied from the Azure portal.
+1. Find the key `Domain` and replace the existing value with your Azure AD tenant name.
+1. Find the key `SignUpSignInPolicyId` and replace with the name of the `Sign up and sign in` policy you created.
+1. Find the key `ResetPasswordPolicyId` and replace with the name of the `Password reset` policy you created.
+1. Find the key `EditProfilePolicyId` and replace with the name of the `Profile editing` policy you created.
+
+### Update the registration for the Web app (WebApp-calls-API-blazor-server)
+
+1. In **App registrations** page, find the *WebApp-calls-API-blazor-server* app.
+1. In the app's registration screen, select the **Certificates & secrets** blade in the left to open the page where we can generate secrets and upload certificates.
+1. In the **Client secrets** section, select **New client secret**:
+   - Type a key description (for instance `app secret`),
+   - Select one of the available key durations (**In 1 year**, **In 2 years**, or **Never Expires**) as per your security posture.
+   - The generated key value will be displayed when you select the **Add** button. Copy the generated value for use in the steps later.
+   - You'll need this key later in your code's configuration files. This key value will not be displayed again, and is not retrievable by any other means, so make sure to note it from the Azure portal before navigating to any other screen or blade.
+1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+   - Select the **Add a permission** button and then,
+   - Ensure that the **My APIs** tab is selected.
+   - In the list of APIs, select the API `ToDoListService-aspnetcore`.
+   - In the **Delegated permissions** section, select the **Access 'ToDoListService-aspnetcore'** in the list. Use the search box if necessary.
+   - Select the **Add permissions** button at the bottom.
+
+#### Configure the Web app (WebApp-calls-API-blazor-server) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open `WebApp-your-API\B2C\Client\appsettings.json` file and copy the keys from "AzureAd" section of previous chapter's `WebApp-OIDC\B2C\blazorserver-B2C\appsettings.json` file.
+1. Find the key `ClientSecret` and replace the existing value with the key you saved during the creation of the `WebApp-calls-API-blazor-server` app, in the Azure portal.
+1. Find the key `TodoListScope` and replace the existing value with Scope.
```

Comment:
carefully write this line to indicate that the developer uses the scope they defined. This is also an issue with Codegen